examples/rag_agent/agent/agent.py

```
@@ -0,0 +1,318 @@
+        """Fallback to web search when no relevant documents found"""
+        try:
+            # Simple: run DDG search and summarize with the LLM
+            ddg = DuckDuckGoSearchRun(num_results=5)
+            hits_text = ddg.run(question)
+            prompt = ChatPromptTemplate.from_messages([
+                ("system", "You are a careful researcher. Summarize the findings concisely and cite sources inline if available."),
+                ("human", "Question: {q}\n\nSearch Results:\n{hits}\n\nWrite a short, factual answer:")
+            ])
+            chain = prompt | self.llm | StrOutputParser()
+            return chain.invoke({"q": question, "hits": hits_text})
+        except Exception:
```

Ruff (0.14.3)

Do not catch blind exception: `Exception`

(BLE001)